packages/e2e-test-utils/src/inserter.js

```diff
@@ -100,157 +142,208 @@ async function waitForInserterCloseAndContentFocus() {
+ * null if nothing was found.
  */
-async function waitForInserterSearch() {
+export async function searchGlobalInserter( category, searchTerm ) {
+	await openGlobalBlockInserter();
+	await page.waitForSelector( INSERTER_SEARCH_SELECTOR );
+	await page.focus( INSERTER_SEARCH_SELECTOR );
+	await pressKeyWithModifier( 'primary', 'a' );
+	await page.keyboard.type( searchTerm );
+
+	// Wait for the default block list to disappear to prevent its items from
+	// being considered as search results. This is needed since we're debouncing
+	// search request.
+	await page.waitForSelector( '.block-editor-inserter__block-list', {
+		hidden: true,
+	} );
+
+	let waitForInsertElement;
+	let waitForNoResults;
+
+	switch ( category ) {
+		case 'Blocks':
+		case 'Patterns':
+		case 'Reusable': {
+			waitForInsertElement = async () => {
+				return await page.waitForXPath(
+					`//*[@role='option' and contains(., '${ searchTerm }')]`
+				);
+			};
+			waitForNoResults = async () => {
+				await page.waitForSelector(
+					'.block-editor-inserter__no-results'
+				);
+				return null;
+			};
+			break;
+		}
+		case 'Block Directory': {
+			waitForInsertElement = async () => {
+				// Return the first item from the Block Directory search results.
+				return await page.waitForSelector(
+					'.block-directory-downloadable-blocks-list button:first-child'
+				);
+			};
+			waitForNoResults = async () => {
+				// Use a soft timeout if Block Directory doesn't return anything
+				// within 5 seconds, as there's no "empty results" element being
+				// rendered when nothing is found.
+				return await new Promise( ( resolve ) =>
+					setTimeout( () => resolve( null ), 5000 )
+				);
+			};
+		}
+	}
+
+	return await Promise.race( [ waitForInsertElement(), waitForNoResults() ] );
+}
+
+/**
+ * Inserts an entity matching given category and term via the global inserter.
+ * If the entity is not instantly available in the open inserter, a search will
+ * be performed. If the search returns no results, an error will be thrown.
+ *
+ * Available categories: Blocks, Patterns, Reusable and Block Directory.
+ *
+ * @param {string} category   The category to insert from.
+ * @param {string} searchTerm The text to search the inserter for.
+ */
+export async function insertFromGlobalInserter( category, searchTerm ) {
+	await openGlobalBlockInserter();
+	await selectGlobalInserterTab( category );
+
+	let insertButton;
+
+	// Check if the block is instantly insertable...
 	try {
+		insertButton = (
+			await page.$x(
+				`//*[@role='option' and contains(., '${ searchTerm }')]`
+			)
+		 )[ 0 ];
+	} catch ( error ) {
+		// noop
+	}
+
+	// ...and if not, search for it.
+	if ( ! insertButton ) {
+		insertButton = await searchGlobalInserter( category, searchTerm );
+	}
+
+	// Throw an error if nothing was found.
+	if ( ! insertButton ) {
+		throw new Error(
+			`Couldn't find "${ searchTerm }" in the ${ category } category.`
```

Comment:
Should we make this message a bit more detailed, by adding that we're performing the search particularly in the inserter?

Reply:
I assumed this text would suffice because it's always paired with the error stack, which points directly to the `insertFromGlobalInserter` method, for example:

<img width="1020" alt="Screenshot 2022-12-16 at 11 37 48" src="https://user-images.githubusercontent.com/1451471/208082920-d14577d5-8172-41bf-a654-3297ca7a5ee6.png">

Does that look OK to you?

Reply:
Sure 👍 I believe there's value in more detailed error messages, but also the context around is important, so I wouldn't mind that.